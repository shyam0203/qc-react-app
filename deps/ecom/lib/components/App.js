'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.App = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CategoryList = require('./CategoryList');

var _CategoryList2 = _interopRequireDefault(_CategoryList);

var _ProductList = require('./ProductList');

var _ProductList2 = _interopRequireDefault(_ProductList);

var _Cart = require('./Cart');

var _Cart2 = _interopRequireDefault(_Cart);

var _Search = require('./Search');

var _Search2 = _interopRequireDefault(_Search);

var _reactRedux = require('react-redux');

var _actions = require('../store/actions');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var App = exports.App = function (_React$PureComponent) {
  (0, _inherits3.default)(App, _React$PureComponent);

  function App() {
    (0, _classCallCheck3.default)(this, App);
    return (0, _possibleConstructorReturn3.default)(this, (App.__proto__ || (0, _getPrototypeOf2.default)(App)).apply(this, arguments));
  }

  (0, _createClass3.default)(App, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        _react2.default.Fragment,
        null,
        _react2.default.createElement(_Search2.default, null),
        _react2.default.createElement(
          'div',
          { id: 'container', style: { display: 'flex' } },
          _react2.default.createElement(
            'div',
            {
              id: 'leftSidebar',
              style: { flex: 3, borderRight: 'thin solid #aaa', paddingRight: 10 } },
            _react2.default.createElement(_CategoryList2.default, { onCategoryClick: this.props.showProductsForCategory })
          ),
          _react2.default.createElement(
            'div',
            { id: 'main', style: { flex: 5 } },
            _react2.default.createElement(_ProductList2.default, null)
          ),
          _react2.default.createElement(
            'div',
            {
              id: 'rightSidebar',
              style: {
                flex: 2,
                display: this.props.showCart ? 'inline-block' : 'none'
              } },
            'The Cart',
            _react2.default.createElement(
              'button',
              { onClick: function onClick() {
                  return _this2.setState({ showCart: false });
                } },
              '=>'
            ),
            _react2.default.createElement(_Cart2.default, null)
          )
        )
      );
    }
  }]);
  return App;
}(_react2.default.PureComponent);

var mapStateToProps = function mapStateToProps(state) {
  return {
    showCart: state.showCart
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, { fetchCategories: _actions.fetchCategories, showProductsForCategory: _actions.showProductsForCategory })(App);