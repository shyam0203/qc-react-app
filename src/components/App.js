import React from 'react';

import CategoryList from './CategoryList';
import ProductList from './ProductList';

import apiClient from '../store/actions/api-client';

class App extends React.PureComponent {
  state = {
    categories: [],
    products: [],
    showCart: false,
    activeCategoryId: '5abbdfde3c43e500a9a3308d',
  };
  async componentDidMount() {
    const categories = await apiClient.getCategories();
    this.setState({ categories });
  }
  showProductsForCategory = async (categoryId) => {
    const products = await apiClient.getProducts(categoryId);
    this.setState({ products, activeCategoryId: categoryId });
  };
  render() {
    return (
      <div id="container" style={{ display: 'flex' }}>
        <div id="leftSidebar" style={{ flex: 3, borderRight: 'thin solid #aaa', paddingRight: 10 }}>
          <CategoryList
            categories={this.state.categories}
            activeCategoryId={this.state.activeCategoryId}
            onCategoryClick={this.showProductsForCategory}
          />
        </div>
        <div id="main" style={{ flex: 5 }}>
          <ProductList products={this.state.products} />
        </div>
        <div
          id="rightSidebar"
          style={{
            flex: 2,
            display: this.state.showCart ? 'inline-block' : 'none',
          }}>
          The Cart......
        </div>
      </div>
    );
  }
}

export default App;
