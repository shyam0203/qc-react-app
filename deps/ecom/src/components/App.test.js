import React from 'react';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { App } from './App';

import { shallow } from 'enzyme';

describe('App', () => {
  it('works', () => {
    const tree = shallow(
      <App />
    );

    expect(tree).toMatchSnapshot();
  });
});
