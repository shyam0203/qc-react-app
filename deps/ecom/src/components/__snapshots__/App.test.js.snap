// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App works 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Search) />,
        <div
          id="container"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            id="leftSidebar"
            style={
              Object {
                "borderRight": "thin solid #aaa",
                "flex": 3,
                "paddingRight": 10,
              }
            }
          >
            <Connect(CategoryList)
              onCategoryClick={undefined}
            />
          </div>
          <div
            id="main"
            style={
              Object {
                "flex": 5,
              }
            }
          >
            <Connect(ProductList) />
          </div>
          <div
            id="rightSidebar"
            style={
              Object {
                "display": "none",
                "flex": 2,
              }
            }
          >
            The Cart
            <button
              onClick={[Function]}
            >
              =&gt;
            </button>
            <Connect(Cart) />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="leftSidebar"
              style={
                Object {
                  "borderRight": "thin solid #aaa",
                  "flex": 3,
                  "paddingRight": 10,
                }
              }
            >
              <Connect(CategoryList)
                onCategoryClick={undefined}
              />
            </div>,
            <div
              id="main"
              style={
                Object {
                  "flex": 5,
                }
              }
            >
              <Connect(ProductList) />
            </div>,
            <div
              id="rightSidebar"
              style={
                Object {
                  "display": "none",
                  "flex": 2,
                }
              }
            >
              The Cart
              <button
                onClick={[Function]}
              >
                =&gt;
              </button>
              <Connect(Cart) />
            </div>,
          ],
          "id": "container",
          "style": Object {
            "display": "flex",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(CategoryList)
                onCategoryClick={undefined}
              />,
              "id": "leftSidebar",
              "style": Object {
                "borderRight": "thin solid #aaa",
                "flex": 3,
                "paddingRight": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onCategoryClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ProductList) />,
              "id": "main",
              "style": Object {
                "flex": 5,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "The Cart",
                <button
                  onClick={[Function]}
                >
                  =&gt;
                </button>,
                <Connect(Cart) />,
              ],
              "id": "rightSidebar",
              "style": Object {
                "display": "none",
                "flex": 2,
              },
            },
            "ref": null,
            "rendered": Array [
              "The Cart",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "=>",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "=>",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Search) />,
          <div
            id="container"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              id="leftSidebar"
              style={
                Object {
                  "borderRight": "thin solid #aaa",
                  "flex": 3,
                  "paddingRight": 10,
                }
              }
            >
              <Connect(CategoryList)
                onCategoryClick={undefined}
              />
            </div>
            <div
              id="main"
              style={
                Object {
                  "flex": 5,
                }
              }
            >
              <Connect(ProductList) />
            </div>
            <div
              id="rightSidebar"
              style={
                Object {
                  "display": "none",
                  "flex": 2,
                }
              }
            >
              The Cart
              <button
                onClick={[Function]}
              >
                =&gt;
              </button>
              <Connect(Cart) />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="leftSidebar"
                style={
                  Object {
                    "borderRight": "thin solid #aaa",
                    "flex": 3,
                    "paddingRight": 10,
                  }
                }
              >
                <Connect(CategoryList)
                  onCategoryClick={undefined}
                />
              </div>,
              <div
                id="main"
                style={
                  Object {
                    "flex": 5,
                  }
                }
              >
                <Connect(ProductList) />
              </div>,
              <div
                id="rightSidebar"
                style={
                  Object {
                    "display": "none",
                    "flex": 2,
                  }
                }
              >
                The Cart
                <button
                  onClick={[Function]}
                >
                  =&gt;
                </button>
                <Connect(Cart) />
              </div>,
            ],
            "id": "container",
            "style": Object {
              "display": "flex",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(CategoryList)
                  onCategoryClick={undefined}
                />,
                "id": "leftSidebar",
                "style": Object {
                  "borderRight": "thin solid #aaa",
                  "flex": 3,
                  "paddingRight": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onCategoryClick": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ProductList) />,
                "id": "main",
                "style": Object {
                  "flex": 5,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "The Cart",
                  <button
                    onClick={[Function]}
                  >
                    =&gt;
                  </button>,
                  <Connect(Cart) />,
                ],
                "id": "rightSidebar",
                "style": Object {
                  "display": "none",
                  "flex": 2,
                },
              },
              "ref": null,
              "rendered": Array [
                "The Cart",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "=>",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "=>",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
